.github/workflows/main.yml

```diff
@@ -20,21 +20,22 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        node: [14]
+        node: [14, 16]
 
     steps:
       - name: Clone repository
         uses: actions/checkout@v2
         with:
           fetch-depth: 3
 
-      - name: Cache multiple paths
-        uses: actions/cache@v2
+      - uses: actions/setup-node@v2
         with:
-          path: |
-            ~/cache
-            !~/cache/exclude
-          key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}
+          node-version: ${{ matrix.node }}
+          cache: 'npm'
+          cache-dependency-path: '**/package-lock.json'
```

Comment:
setup-node support cache option that use actions/cache internally.

https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-dependencies

Reply:
cool I didn't know about this setup-node action, looks like a nice one to use